apps/hash-graph/bin/cli/src/parser.rs

```diff
@@ -0,0 +1,29 @@
+use std::{ffi::OsStr, str::FromStr};
+
+use clap::{builder::TypedValueParser, error::ErrorKind, Arg, Command, Error};
+
+#[derive(Clone)]
+pub struct OptionalSentryDsnParser;
+
+impl TypedValueParser for OptionalSentryDsnParser {
+    type Value = Option<sentry::types::Dsn>;
+
+    fn parse_ref(
+        &self,
+        cmd: &Command,
+        _: Option<&Arg>,
+        value: &OsStr,
+    ) -> Result<Self::Value, Error> {
+        if value.is_empty() {
+            Ok(None)
+        } else {
+            let Some(value) = value.to_str() else {
+                return Err(Error::new(ErrorKind::InvalidValue).with_cmd(cmd));
+            };
+
+            sentry::types::Dsn::from_str(value)
+                .map(Some)
+                .map_err(|_error| Error::new(ErrorKind::InvalidValue))
+        }
+    }
+}
```

Comment:
For some reason the CI fails as it interprets this parameter as required.

Reply:
This is because we use `core::option::Option<T>`, instead of `Option<T>`, which means that clap does not set it to `required = false`. We need to do this manually.

fixed in [6e591ab](https://github.com/hashintel/hash/pull/3091/commits/6e591ab5bb600a6b7881131b437fb0e044584f83)